ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -1494,31 +1620,25 @@ FEDataManager::interpWeighted(const int f_data_idx,
                 }
 
                 // JxW depends on the element
-                fe_map.compute_map(dim, qrule.get_weights(), elem, /*second derivatives*/false);
-                const std::vector<double> &JxW_F = fe_map.get_JxW();
+                fe_map.compute_map(dim, qrule.get_weights(), elem, /*second derivatives*/ false);
+                const std::vector<double>& JxW_F = fe_map.get_JxW();
                 const std::vector<std::vector<double> >& phi_F = F_fe.get_phi();
 
                 const unsigned int n_qp = qrule.n_points();
                 TBOX_ASSERT(n_qp == phi_F[0].size());
                 TBOX_ASSERT(n_qp == JxW_F.size());
                 const size_t n_basis = F_dof_indices[0].size();
-                for (unsigned int qp = 0; qp < n_qp; ++qp)
-                {
-                    const int idx = n_vars * (qp_offset + qp);
-                    for (unsigned int k = 0; k < n_basis; ++k)
-                    {
-                        const double p_JxW_F = phi_F[k][qp] * JxW_F[qp];
-                        for (unsigned int i = 0; i < n_vars; ++i)
-                        {
-                            F_rhs_e[i](k) += F_qp[idx + i] * p_JxW_F;
-                        }
-                    }
-                }
+                integrate_cell_rhs(n_vars, n_basis, qp_offset, phi_F, JxW_F, F_qp, F_rhs_concatenated);
+
                 for (unsigned int i = 0; i < n_vars; ++i)
                 {
+                    std::copy(F_rhs_concatenated.begin() + i * n_basis,
```

Comment:
Is it worth making a helper function for this?

Reply:
(Presumably we will want to do this kind of thing in other places?)

Reply:
You mean with the copy?

Reply:
Yeah, going between interleaved and non-interleaved forms.

Reply:
Since this is a one-liner and not applicable to spreading (and based on your comment below) I think its okay to hold off on putting this in a helper function.